### PR TITLE
Add ci-scripts

### PR DESCRIPTION
**Pull request by dbasunag**
jira-ticket: https://issues.redhat.com/browse/CNV-39285

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: .gitlint
Comment (jpeimer) [inline .gitlint:1]: Is it supposed to be empty? Do we need it?

File: ci_scripts/utils.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    return params_dict
+
+
+def print_status(status_dict):
+    for key, value in status_dict.items():
+        print(f"    {key}:  {' '.join(list(set(value)))}")
+    print("\n")
+
+
+def all_python_files():
```
Comment (jpeimer) [inline ci_scripts/utils.py:23]: maybe `get_all_python_files` ?

File: ci_scripts/jira_scripts/check_jira_status.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    )
+    _is_jira_open = re.findall(rf"jira_id\s*=[\s*\"\']*{issue_pattern}.*", file_content)
+    _jira_url_jiras = re.findall(
+        rf"https://issues.redhat.com/browse/{issue_pattern}.*",
+        file_content,
+    )
+    return set(_pytest_jira_marker_bugs + _is_jira_open + _jira_url_jiras)
+
+
+def get_jiras_from_python_files():
```
Comment (jpeimer) [inline ci_scripts/jira_scripts/check_jira_status.py:82]: `get_jiras_from_all_python_files` ?

File: ci_scripts/jira_scripts/check_jira_status.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            try:
+                jira_metadata = get_jira_metadata(
+                    jira_id=jira_id, jira_connection=jira_connection
+                )
+                current_jira_status = jira_metadata.status.name.lower()
+                if current_jira_status in closed_statuses:
+                    closed_jiras.setdefault(filename, []).append(
+                        f"{jira_id} [{current_jira_status}]"
+                    )
+                jira_target_release_version = get_jira_fix_version(
```
Comment (jpeimer) [inline ci_scripts/jira_scripts/check_jira_status.py:110]: Should we check the fix version for bugs only? Sometimes we use jira ids of Epics / Stories / Tasks - we do not necessarily own them, so we can't make the clones targeter to other versions.
And, for example, if there's a task to fix some automation - to me it looks enough to state in the acceptance criteria that it should be backported, and there's no need to clone the Task just to give it a matching fix version.

File: ci_scripts/utils.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import os
+from configparser import ConfigParser
+from pathlib import Path
+
+
+def get_connection_params():
```
Comment (jpeimer) [inline ci_scripts/utils.py:6]: `get_jira_connection_params` ?

File: tox.ini [hunk 5]
```diff
@@ -1,11 +1,55 @@
 [tox]
-skipsdist = True
+envlist=code-check
```
Comment (jpeimer) [inline tox.ini:2]: should we add `verify-tc-requirement-polarion`,`verify-bugs-are-open` ?

Comment (jpeimer) [inline tox.ini:2]: WE also had a `pytest-check`, we want to add it later?

File: tox.ini [hunk 6]
```diff
@@ -1,11 +1,55 @@
...
-[testenv]
-setenv = PYTHONPATH = {toxinidir}
-deps =
+[testenv:code-check]
+basepython = python3
+recreate=True
+setenv =
+    PYTHONPATH = {toxinidir}
+deps=
     pre-commit
```
Comment (jpeimer) [inline tox.ini:11]: do we need a `gitlint` as well?

**Review by jpeimer — CHANGES_REQUESTED** [review]

File: ci_scripts/jira_scripts/check_jira_status.py (hunk 3 above)
Comment (dbasunag) [reply ci_scripts/jira_scripts/check_jira_status.py:110]: No. We should not put any tasks/stories/epics to our test automation, specially in non-relevant branches. This should be tracked in jira backlogs only.

File: tox.ini (hunk 6 above)
Comment (dbasunag) [reply tox.ini:11]: Pre-commit etc. are now handled via webhook (Meni's) we no longer need it here.

File: tox.ini (hunk 5 above)
Comment (dbasunag) [reply tox.ini]: yes, we will add it, after we add some tests. This work would be ongoing and iterative.

File: ci_scripts/jira_scripts/check_jira_status.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        token_auth=connection_params.get("token"),
+        options={"server": connection_params.get("url")},
+    )
+    return jira_connection
+
+
+def get_jira_metadata(jira_id, jira_connection):
+    retries = 0
+    max_retry = 3
+    while True:
```
Comment (jpeimer) [inline ci_scripts/jira_scripts/check_jira_status.py:24]: while retries < max_retry:

File: ci_scripts/jira_scripts/check_jira_status.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        try:
+            return jira_connection.issue(
+                id=jira_id, fields="status, issuetype, fixVersions"
+            ).fields
+        except JIRAError as e:
+            # Check for inactivity error (adjust based on your library)
+            if "Unauthorized" in str(e) or "Session timed out" in str(e):
+                retries += 1
+                print(
+                    "Failed to get issue due to inactivity, retrying (%d/%d)",
```
Comment (jpeimer) [inline ci_scripts/jira_scripts/check_jira_status.py:34]: can be f string

File: ci_scripts/jira_scripts/check_jira_status.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+
+def get_jira_metadata(jira_id, jira_connection):
+    retries = 0
+    max_retry = 3
+    while True:
+        try:
+            return jira_connection.issue(
+                id=jira_id, fields="status, issuetype, fixVersions"
+            ).fields
+        except JIRAError as e:
```
Comment (jpeimer) [inline ci_scripts/jira_scripts/check_jira_status.py:29]: except JIRAError as jira_exception:

**Review by jpeimer — CHANGES_REQUESTED** [review]

**Review by jpeimer — APPROVED** [review]

**Pull request merged**